### PR TITLE
Bleach stubs: fix typos, use protocol classes

### PR DESCRIPTION
**Pull request by Deimos**
I mentioned [in a comment a few weeks ago](https://github.com/python/typeshed/pull/2709#issuecomment-483120729) that I was hitting some errors on my codebase from the new Bleach stubs. This was mostly due to the stubs being very specific and using `List[]` for various arguments. These arguments are generally used for `x in y` checks, so I switched all of them to use the `Container[]` protocol class instead.

There were also a couple of other minor, similar changes, and I fixed a couple of typos (in a separate commit to make it easier to see for review, but they could be squashed together if you like).

(I'll ping @srittau since he wrote the original stubs)

### CODE REVIEW

File: third_party/2and3/bleach/sanitizer.pyi [hunk 1]
```diff
@@ -1,7 +1,7 @@
-from typing import List, Dict, Any, Optional, Type, Pattern, Union, Callable, Container, Text
+from typing import Any, Callable, Container, Dict, Iterable, List, Optional, Pattern, Text, Type, Union
 
 ALLOWED_TAGS: List[Text]
-ALLOWED_ATTRIBUTES: Dict[Text, List[Text]]
+ALLOWED_ATTRIBUTES: Dict[Text, Container[Text]]
```
Comment (srittau) [inline third_party/2and3/bleach/sanitizer.pyi]: Why did you change this type? All the other constants use concrete types and this seems more appropriate here as well.

Comment (Deimos) [reply third_party/2and3/bleach/sanitizer.pyi]: Oh, I'm not sure. You're right, that should probably stay as `List`. I pushed another commit reverting that now.

Do you want me to squash these into a single commit?

Comment (srittau) [reply third_party/2and3/bleach/sanitizer.pyi]: Thanks! We prefer not squashing for easier reviewability. We will squash when merging.
